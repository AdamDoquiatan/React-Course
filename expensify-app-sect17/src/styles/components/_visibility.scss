.show-for-mobile {
  @media (min-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-for-desktop {
  @media (max-midth: $desktop-breakpoint - .01rem) {
    display: none;
  }
}