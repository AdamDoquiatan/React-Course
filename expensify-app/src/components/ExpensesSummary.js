import React from 'react'
import ReactDOM from 'react-dom'
import { connect } from 'react-redux'
import numeral from 'numeral'
import getExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'

export const ExpensesSummery = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  return (
    <div>
      <p>Viewing {expenseCount} {expenseWord} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
    </div>
)
}

const mapStateToProps = (state, props) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
  }
}

export default connect(mapStateToProps)(ExpensesSummery)