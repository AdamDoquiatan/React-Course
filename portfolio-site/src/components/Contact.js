import React from 'react'

const Contact = () => {
  return (
    <div>
      <h1>Contact me!</h1>
      <p>Contact me at adamdoq@gmail.com</p>
    </div>
  )
}

export default Contact