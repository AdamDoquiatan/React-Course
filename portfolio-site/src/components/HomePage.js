import React from 'react'

const HomePage = () => {
  return (
    <div>
      <h1>Welcome</h1>
      <p>This is my Site!</p>
    </div>
  )
}

export default HomePage