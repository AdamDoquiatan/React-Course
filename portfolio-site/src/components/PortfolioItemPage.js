import React from 'react'

const PortfolioItem = (props) => (
    <div>
      <h1>A thing I've done</h1>
      <p>This is my project {props.match.params.id}</p>
    </div>

)

export default PortfolioItem