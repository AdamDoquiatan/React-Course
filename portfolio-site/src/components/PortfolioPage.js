import React from 'react'
import { Link } from 'react-router-dom'

const PortfolioPage = () => {
  return (
    <div>
      <h1>My Work</h1>
      <p>Check out the following things I've done!</p>
      <Link to='/portfolio/1'>Project 1</Link>
      <Link to='/portfolio/2'>Project 2</Link>
    </div>
  )
}

export default PortfolioPage

