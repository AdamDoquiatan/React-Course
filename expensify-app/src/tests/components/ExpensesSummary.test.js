import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummery } from '../../components/ExpensesSummary'
import selectExpenses from '../../selectors/expenses'
import getExpensesTotal from '../../selectors/expenses-total'
import expenses from '../fixtures/expenses'
import filters from '../fixtures/filters'

test('should render ExpensesSummery correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummery
    expenseCount={1} 
    expensesTotal={235}
  />)
})

test('should render ExpensesSummery correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummery
    expenseCount={23} 
    expensesTotal={54325}
  />)
  expect(wrapper).toMatchSnapshot()
})

