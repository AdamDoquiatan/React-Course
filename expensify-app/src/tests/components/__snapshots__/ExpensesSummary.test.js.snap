// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummery correctly with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     totalling 
    $543.25
  </p>
</div>
`;
